airflow/plugins_manager.py

```diff
@@ -142,32 +142,31 @@ def is_valid_plugin(plugin_obj, existing_plugins):
 # Crawl through the plugins folder to find AirflowPlugin derivatives
 for root, dirs, files in os.walk(settings.PLUGINS_FOLDER, followlinks=True):
     for f in files:
+        filepath = os.path.join(root, f)
         try:
-            filepath = os.path.join(root, f)
-            try:
-                if not os.path.isfile(filepath):
-                    continue
-                mod_name, file_ext = os.path.splitext(
-                    os.path.split(filepath)[-1])
-                if file_ext != '.py':
-                    continue
-
-                log.debug('Importing plugin module %s', filepath)
-                # normalize root path as namespace
-                namespace = '_'.join([re.sub(norm_pattern, '__', root), mod_name])
-
-                m = imp.load_source(namespace, filepath)
-                for obj in list(m.__dict__.values()):
-                    if is_valid_plugin(obj, plugins):
-                        plugins.append(obj)
-            except Exception as e:  # pylint: disable=broad-except
-                log.exception(e)
-                log.error('Failed to import plugin %s', filepath)
-                import_errors[filepath] = str(e)
+            if not os.path.isfile(filepath):
+                continue
+            mod_name, file_ext = os.path.splitext(
+                os.path.split(filepath)[-1])
+            if file_ext != '.py':
+                continue
+
+            log.debug('Importing plugin module %s', filepath)
+            # normalize root path as namespace
+            namespace = '_'.join([re.sub(norm_pattern, '__', root), mod_name])
+
+            m = imp.load_source(namespace, filepath)
+            for obj in list(m.__dict__.values()):
+                if is_valid_plugin(obj, plugins):
+                    plugins.append(obj)
         except Exception as e:  # pylint: disable=broad-except
             log.exception(e)
-            log.error('Failed to import plugin %s', str(f))
-            import_errors[str(f)] = str(e)
+            if not filepath:
+                log.error('Failed to import plugin %s', str(f))
+                import_errors[str(f)] = str(e)
+            else:
+                log.error('Failed to import plugin %s', filepath)
+                import_errors[filepath] = str(e)
```

Comment:
```python
                filepath = filepath or str(f)
                log.error('Failed to import plugin %s', filepath)
                import_errors[filepath] = str(e)
```
isn't this better?

Reply:
I am going to change it in follow-up request, but yes.

Reply:
If there is nothing else to do you can leave it. :)

Reply:
I changed it :)